code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -52,12 +52,124 @@
 	push_data()
 	activate_pin(2)
 
+/obj/item/integrated_circuit/list/search
+	name = "search circuit"
+	desc = "This circuit will give index of desired element in the list."
+	extended_desc = "Search will start at 1 position and will return first matching position"
+	inputs = list(
+		"list" = IC_PINTYPE_LIST,
+		"item" = IC_PINTYPE_ANY
+	)
+	outputs = list(
+		"index" = IC_PINTYPE_NUMBER
+	)
+	icon_state = "addition"
+	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
+
+/obj/item/integrated_circuit/list/search/do_work()
+	var/list/input_list = get_pin_data(IC_INPUT, 1)
+	var/item = get_pin_data(IC_INPUT, 2)
+	set_pin_data(IC_OUTPUT, 1, input_list.Find(item))
+	push_data()
+	activate_pin(2)
+
+/obj/item/integrated_circuit/list/at
+	name = "at circuit"
+	desc = "This circuit will pick element from the list by number"
+	extended_desc = "If there is no element with such number, result will be null"
+	inputs = list(
+		"list" = IC_PINTYPE_LIST,
+		"index" = IC_PINTYPE_NUMBER
+	)
+	outputs = list("item" = IC_PINTYPE_ANY)
+	icon_state = "addition"
+	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
+
+/obj/item/integrated_circuit/list/at/do_work()
+	var/list/input_list = get_pin_data(IC_INPUT, 1)
+	var/index = get_pin_data(IC_INPUT, 2)
+	var/item = input_list[index]
+	set_pin_data(IC_OUTPUT, 1, item)
+	push_data()
+	activate_pin(2)
+
+/obj/item/integrated_circuit/list/delete
+	name = "delete circuit"
+	desc = "This circuit will pick element from the list by number"
```

Comment:
Missing period.

Reply:
`This circuit will delete the element from a list by index.`